pkg/onepassword/volumes.go

```diff
@@ -24,6 +38,17 @@ func AppendUpdatedVolumeSecrets(volumes []corev1.Volume, secrets map[string]*cor
 				updatedDeploymentSecrets[secret.Name] = secret
 			}
 		}
+		if volumes[i].Projected != nil {
+			for j := 0; j < len(volumes[i].Projected.Sources); j++ {
+				if secret := volumes[i].Projected.Sources[j].Secret; secret != nil {
+					secretName := secret.Name
+					secret, ok := secrets[secretName]
+					if ok {
+						updatedDeploymentSecrets[secret.Name] = secret
```

Comment:
I see we use this logic in a number of places now, maybe we can extract it into a separate method for cleaner code.